src/Exception.php

```diff
@@ -406,10 +168,58 @@ public function getParams()
      *
      * @return $this
      */
-    public function addMoreInfo($param, $value)
+    public function addMoreInfo($param, $value): self
     {
         $this->params[$param] = $value;
 
         return $this;
     }
+
+    /**
+     * Add a suggested/possible solution to the exception.
+     *
+     * @TODO can be added more features? usually we are out of App
+     *
+     * @param string $solution
+     *
+     * @return Exception
+     */
+    public function addSolution(string $solution)
+    {
+        $this->solutions[] = $solution;
+
+        return $this;
+    }
+
+    /**
+     * Get the solutions array.
+     */
+    public function getSolutions(): array
+    {
+        return $this->solutions;
+    }
+
+    /**
+     * Get the custom Exception name, if defined in $custom_exception_name.
+     *
+     * @return string
+     */
+    public function getCustomExceptionName(): string
+    {
+        if ($this->custom_exception_name) {
+            return $this->custom_exception_name;
+        }
+
+        return get_class($this);
```

Comment:
Since atk4/core is targetting 7.2+ why not just 
`return $this->custom_exception_name ?? get_class($this);`

Reply:
right!